File: AdvancePython/Day-3 Advanced OOP concepts/Task-3-2.py
# Task 3:
# 2. Constructors:
def __init__(self) -> None:
    # body of the constructor
    pass


class Parent:
    def __init__(self, item: list[int]) -> str:
        self.item: list[int] = item
        print(f"Parent class: {self.item}")


class SubClass(Parent):
    def __init__(self, item: list[int]) -> None:
        super().__init__(item)
        print(f"Output of SubClass")


data = [1, 2, 3, 4, 5]
instance = SubClass(data)

File: node.js/Day-3/Task-3-2.js
// Task 3:
// 2. Constructors:
class Parent {
    constructor(...args) {
        console.log(`Parent's class ${args}`);
        this.data = x;
    }
}

class SubClass extends Parent {
    constructor(...args) {
        super(...args);
    }
}

const newclass = new SubClass(1, 2, 3, 4);
console.log(newclass);